### PR TITLE
rpm: move mount.ceph from ceph-base to ceph-common and add symlink in /sbin for SUSE

### PR DESCRIPTION
**Pull request by smithfarm**
Fixes: http://tracker.ceph.com/issues/16598
Signed-off-by: Nathan Cutler <ncutler@suse.com>

### CODE REVIEW

**Comment by smithfarm** [comment]
@ktdreyer @b-ranto @tserong Care to review?

**Comment by b-ranto** [comment]
@smithfarm I would not do this in the %post phase. You will leave the sym-link there even when the user decides to remove the package.

**Comment by smithfarm** [comment]
@b-ranto Thanks. This is the first time I've packaged a symlink. How about now?

**Comment by b-ranto** [comment]
lgtm

**Comment by smithfarm** [comment]
gitbuilders are ok with it

**Comment by smithfarm** [comment]
`teuthology-suite --verbose --suite upgrade/hammer-x --suite-branch master --ceph wip-16598 --machine-type vps --priority 101 --filter centos_7 --limit 1 machine_types/vps.yaml`

**running** http://pulpito.ceph.com/smithfarm-2016-07-07_05:21:59-upgrade:hammer-x-wip-16598---basic-vps/

**Comment by tserong** [comment]
LGTM

**Comment by smithfarm** [comment]
teuthology is down atm

**Comment by ktdreyer** [comment]
Good catch Boris. LGTM now

**Comment by smithfarm** [comment]
`teuthology-suite --verbose --suite upgrade/jewel-x --suite-branch master --ceph wip-16598 --machine-type vps --priority 101 --filter centos_7 --limit 1 machine_types/vps.yaml`

**running** http://pulpito.ceph.com/smithfarm-2016-07-07_11:07:51-upgrade:jewel-x-wip-16598---basic-vps/

**Comment by smithfarm** [comment]
The `upgrade/hammer-x` test job failed due to a known bug, but the failure is unrelated to this PR.

**Comment by smithfarm** [comment]
test this please

**Comment by smithfarm** [comment]
Hey, jenkins, test this please

**Comment by smithfarm** [comment]
The jenkins failure is 

```
*** Caught signal (Segmentation fault) **
 in thread 2af2c7dfb700 thread_name:tp_async_compr
```

It happens in Test  #72: unittest_async_compressor

**Comment by smithfarm** [comment]
@ktdreyer @b-ranto We just noticed that mount.ceph is in ceph-base - shouldn't it be in ceph-common??? I'm thinking I could add a commit to this PR to move it, and then merge.

**Comment by ktdreyer** [comment]
moving to ceph-common (from ceph-base) sounds right to me.

**Comment by smithfarm** [comment]
Whoa, in debian it's in its own package "ceph-fs-common" (?!)

**Comment by smithfarm** [comment]
And it's been that way for 4 years :-)

https://github.com/ceph/ceph/commit/915f3210962a78913a72fd3c85d84594323d0d30

**Comment by smithfarm** [comment]
Changelog:

* rebase
* add a commit to move mount.ceph (and manpage) from ceph-base to ceph-common

**Comment by b-ranto** [comment]
@smithfarm Could you also patch Makefile-{server,client}.am so that the binary gets build as a client binary? All you need to do there is to move the few lines that define the build from Makefile-server.am to Makefile-client.am (probably under WITH_CEPHFS ifdef).

File: src/Makefile-client.am [hunk 1]
```diff
@@ -77,6 +77,13 @@ cephfs_SOURCES = cephfs.cc
 cephfs_LDADD = $(LIBCOMMON)
 bin_PROGRAMS += cephfs
 
+mount_ceph_SOURCES = mount/mount.ceph.c
+mount_ceph_LDADD = $(LIBSECRET) $(LIBCOMMON)
+if LINUX
+sbin_PROGRAMS += mount.ceph
+endif # LINUX
+sbin_SCRIPTS += mount.fuse.ceph
```
Comment (b-ranto) [inline src/Makefile-client.am]: Hmm, mount.fuse should probably go few lines up to the WITH_FUSE section.

**Comment by b-ranto** [comment]
if it passes the gitbuilfders then it lgtm.

**Comment by smithfarm** [comment]
Changelog:

* Makefile-client.am: moved mount.fuse.ceph up to WITH_FUSE section

**Comment by ktdreyer** [comment]
`Reviewed-by: Ken Dreyer <kdreyer@redhat.com>`

**Pull request merged**